### PR TITLE
Add 0.3.1 on aarch64

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Bug Tracker: new open issues https://github.com/dask/hdfs3/issues
Upstream license: https://github.com/dask/hdfs3/blob/0.3.1/LICENSE.txt
Upstream setup.py:  https://github.com/dask/hdfs3/blob/0.3.1/setup.py

Actions:
1. Add missing packages: `pip`, `wheel`
2. Add pip check
3. Add dev_url
4. Fix the license
5. Skip building on `osx `and `s390x`

Result:

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,30 +10,42 @@ source:
...
+  license: BSD-3-Clause
   license_family: BSD
   license_file: LICENSE.txt
   summary: Python wrapper for libhdfs3
+  # This project is not undergoing development.
+  # Pyarrow's JNI hdfs interface is mature and stable. 
+  # It also has fewer problems with configuration and various security settings, 
+  # and does not require the complex build process of libhdfs3. 
+  # Therefore, all users who have trouble with hdfs3 are recommended to try pyarrow.
+  dev_url: https://github.com/dask/hdfs3
```
Comment (varlackc) [inline recipe/meta.yaml]: We might have too add the `doc_url:`

**Review by varlackc — COMMENTED** [review]
We might need to add the `doc_url`

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Thanks. I added `doc_url: https://hdfs3.readthedocs.io/en/latest/`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,30 +10,43 @@ source:
   sha256: 3be9ca9d812e760969419e7288fef4d6700e7aa01d303aad4e56cd4ecb4f11ed
 
 build:
-  number: 0
+  number: 1
   script: python setup.py install --single-version-externally-managed --record=record.txt
   # This packages uses libhdfs3, which is primarily used with HDFS on Linux.
-  skip: True  # [win]
+  # There is no libhdfs3 >=2.3 and its dependencies on s390x
+  skip: True  # [win or osx or s390x]
```
Comment (tobijk) [inline recipe/meta.yaml:17]: Why skip on osx? I see we have built hdfs3 for osx-64 before?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,30 +10,43 @@ source:
...
 
 requirements:
   host:
     - python
+    - pip
     - setuptools
-
+    - wheel
```
Comment (tobijk) [inline recipe/meta.yaml:24]: Just a suggestion, but pip already depends on wheel, so unless we use wheel directly, it seems we don't need to mention it as a dependency.

Comment (skupr-anaconda) [reply recipe/meta.yaml:24]: Kai recommends explicitly use`wheel` in `host`. And his [crender](https://github.com/katietz/crender) tool recommends the same

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:17]: We haven't `libhdfs3` on osx-64 and it's pretty weird how `hdfs3 0.3.1` was built without our libhdfs3

Comment (skupr-anaconda) [reply recipe/meta.yaml:17]: `libhdfs3` https://anaconda.org/anaconda/libhdfs3/files
`hdfs3` https://anaconda.org/anaconda/hdfs3/files

Comment (skupr-anaconda) [reply recipe/meta.yaml:17]: I removed osx from `skip`.

File: recipe/meta.yaml (hunk 3 above)
Comment (tobijk) [reply recipe/meta.yaml:24]: Hm, ok.

File: recipe/meta.yaml (hunk 2 above)
Comment (tobijk) [reply recipe/meta.yaml:17]: > We haven't libhdfs3 on osx-64 and it's pretty weird how hdfs3 0.3.1 was built without our libhdfs3

That is totally weird, you are right about that. Apparently then we don't have a functional hdf3 on osx, so we may as well skip it. /me shrugs

Comment (skupr-anaconda) [reply recipe/meta.yaml:17]: Rebuilt it with `skip: True  # [win or osx or s390x]`

**Review by tobijk — APPROVED** [review]
LGTM

**Pull request merged**